offer-manager-domain/src/main/groovy/life/qbic/business/products/create/CreateProductDataSource.groovy

```diff
@@ -19,4 +20,12 @@ interface CreateProductDataSource {
      * @throws ProductExistsException if the product already exists in the data source
      */
     void store(Product product) throws DatabaseQueryException, ProductExistsException
+
+    /**
+     * Fetches the latest version of a product identifier for a product category
+     * @param productCategory
+     * @return
+     * @throws DatabaseQueryException
+     */
+    Optional<ProductId> fetchLatestProductIdentifierVersion(ProductCategory productCategory) throws DatabaseQueryException
```

Comment:
I think this should not be part of the API. Your datasource needs to make sure, that a new incremental id is assigned, once a new product is created in the data source.

Reply:
Should we then open an issue for that?

Reply:
I found we already have an issues for this: https://github.com/qbicsoftware/offer-manager-2-portlet/issues/395